### DISCUSSION TITLE
onEdgeLabelRendered渲染react组件，组件内部更新问题

### DISCUSSION THREAD

**Discussion by cuiyiworld** [antvis/X6#1533] [Q&A] [answered]
我在使用onEdgeLabelRendered渲染了一个react组件，
![image](https://user-images.githubusercontent.com/21233144/140945473-71c5deb6-95d4-4e67-bedd-a616233723aa.png)
如图，我更新的边edge的data里面的数据，即react组件参数发生更新，但是react组件内部没有更新。请问怎么才能使组件内部发生更新？

**Comment by NewByVector** [comment, marked as answer]
需要修改 label 才能触发 `onEdgeLabelRendered`

**Reply by komagic** [reply]
关键是label需要根据context 中的状态变化，又没有新增label
